rir/src/compiler/pir/instruction.h

```diff
@@ -1154,25 +1187,28 @@ SIMPLE_INSTRUCTIONS(V, _)
                   {{lhs, rhs}}, env, srcIdx) {}                                \
         void updateType() override final {                                     \
             maskEffectsAndTypeOnNonObjects(Type);                              \
+            maskEffect(PirType::SafeType().notObject(), Effect::Warn);         \
```

Comment:
this confused me a lot. Because i was searching for the SafeType function in PirType. Can you put this on the outside? Basically write `BINOP(Mul, lhs()->type | rhs()->type, PirType::num().notObject())` at the outside. Otherwise it is very confusing, especially given that the other `Type` thing is done in this way. Brevity is not always better, if it decreases readability